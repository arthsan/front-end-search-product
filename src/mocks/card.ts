import { Product } from 'store/useProductStore'

export const cardMock: Product = {
  id: 'newID',
  name: 'Camiseta',
  type: 'organic',
  _meta: {
    score: 10,
    visitsClickCount: '100'
  }
}
