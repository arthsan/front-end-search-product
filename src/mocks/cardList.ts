import { Product } from 'store/useProductStore'

export const cardList: Product[] = [
  {
    name: 'Kit Com 5 Camisetas Básicas Esportivas Masculina Preta, Branca E Cinza',
    id: '56664056',
    _meta: {
      score: 345.77496,
      visitsClickCount: '779'
    },
    type: 'organic'
  },
  {
    name: 'Camiseta masculina jordam 100% algodão',
    id: '4116684846',
    _meta: {
      score: 212.25824,
      visitsClickCount: '246'
    },
    type: 'organic'
  },
  {
    name: 'Kit Com 5 Camisetas Básicas Part. B Masculina Algodão T-shirt Tee',
    id: '80502107',
    _meta: {
      score: 199.59839,
      visitsClickCount: '445'
    },
    type: 'organic'
  },
  {
    name: 'Camiseta Estampada Plus Size Masculina 1000075054 Malwee Wee',
    id: '2533730399',
    _meta: {
      score: 123.03333,
      visitsClickCount: '158'
    },
    type: 'organic'
  },
  {
    name: 'Kit 5 Camisetas Básicas Masculina 100% Poliéster Cores Sortidas',
    id: '3374278226',
    _meta: {
      score: 110.10536,
      visitsClickCount: '131'
    },
    type: 'organic'
  },
  {
    name: 'Camiseta Masculina Personalizada Stranger Things Série',
    id: '2418950425',
    _meta: {
      score: 105.30116,
      visitsClickCount: '183'
    },
    type: 'organic'
  }
]
