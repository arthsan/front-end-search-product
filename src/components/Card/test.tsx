import { screen, render } from '@testing-library/react'

import { Card } from '.'

import { cardMock } from '../../mocks/card'

describe('<Card />', () => {
  it('should render the heading with product title', () => {
    const { container } = render(
      <Card
        name={cardMock.name}
        _meta={cardMock._meta}
        type={cardMock.type}
        key={cardMock.id}
      />
    )
    const heading = screen.getByRole('title')
    expect(heading).toHaveTextContent(cardMock.name)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render correctly the product type', () => {
    const { container } = render(
      <Card
        name={cardMock.name}
        _meta={cardMock._meta}
        type={cardMock.type}
        key={cardMock.id}
      />
    )
    const description = screen.getByRole('description')
    expect(description).toHaveTextContent(cardMock.type)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the product click views', () => {
    const { container } = render(
      <Card
        name={cardMock.name}
        _meta={cardMock._meta}
        type={cardMock.type}
        key={cardMock.id}
      />
    )
    const views = screen.getByRole('views')
    expect(views).toHaveTextContent(cardMock._meta.visitsClickCount)
    expect(container.firstChild).toMatchSnapshot()
  })
})
