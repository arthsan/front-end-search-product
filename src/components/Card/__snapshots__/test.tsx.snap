// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should render correctly the product type 1`] = `
<div
  class="css-njtlda"
>
  <h2
    class="chakra-heading css-1s3enhb"
    role="title"
  >
    Camiseta
  </h2>
  <p
    class="chakra-text css-in3yi3"
    role="description"
  >
    organic
  </p>
  <p
    class="chakra-text css-20qhnm"
  >
    Store Ranking 
    10
  </p>
  <p
    class="chakra-text css-0"
    role="views"
  >
    Public views 
    100
  </p>
</div>
`;

exports[`<Card /> should render the heading with product title 1`] = `
<div
  class="css-njtlda"
>
  <h2
    class="chakra-heading css-1s3enhb"
    role="title"
  >
    Camiseta
  </h2>
  <p
    class="chakra-text css-in3yi3"
    role="description"
  >
    organic
  </p>
  <p
    class="chakra-text css-20qhnm"
  >
    Store Ranking 
    10
  </p>
  <p
    class="chakra-text css-0"
    role="views"
  >
    Public views 
    100
  </p>
</div>
`;

exports[`<Card /> should render the product click views 1`] = `
<div
  class="css-njtlda"
>
  <h2
    class="chakra-heading css-1s3enhb"
    role="title"
  >
    Camiseta
  </h2>
  <p
    class="chakra-text css-in3yi3"
    role="description"
  >
    organic
  </p>
  <p
    class="chakra-text css-20qhnm"
  >
    Store Ranking 
    10
  </p>
  <p
    class="chakra-text css-0"
    role="views"
  >
    Public views 
    100
  </p>
</div>
`;
